edi_oca/models/edi_backend.py

```diff
@@ -379,17 +383,20 @@ def _check_output_exchange_sync(self, skip_send=False, skip_sent=True):
 
         self._exchange_check_ack_needed(pending_records)
 
-    def _output_new_records_domain(self):
+    def _output_new_records_domain(self, record_ids=None):
         """Domain for output records needing output content generation."""
-        return [
+        domain = [
             ("backend_id", "=", self.id),
             ("type_id.exchange_file_auto_generate", "=", True),
             ("type_id.direction", "=", "output"),
             ("edi_exchange_state", "=", "new"),
             ("exchange_file", "=", False),
         ]
+        if record_ids:
+            domain.append(("id", "in", record_ids))
+        return domain
 
-    def _output_pending_records_domain(self, skip_sent=True):
+    def _output_pending_records_domain(self, skip_sent=True, record_ids=None):
```

Comment:
Why do yo do this?

I mean, isn't it used on the cron only? I assume you need it for something else, isn't it?

Reply:
Understood after checking #653